caikit/runtime/train.py

```diff
@@ -0,0 +1,198 @@
+# Copyright The Caikit Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+This module is a central entrypoint for running a single synchronous training
+job using caikit.core.train
+"""
+
+# Standard
+from typing import Type
+import argparse
+import importlib
+import json
+import os
+import sys
+
+# Third Party
+from google.protobuf import json_format
+
+# First Party
+import alog
+
+# Local
+from ..core import ModuleBase, train
+from ..core.data_model import TrainingStatus
+from ..core.exceptions import error_handler
+from ..core.registries import module_registry
+from ..core.toolkit.logging import configure as config_logging
+from .names import get_service_package_name
+from .service_factory import ServicePackageFactory
+from .utils.servicer_util import build_caikit_library_request_dict
+
+log = alog.use_channel("TRAIN")
+error = error_handler.get(log)
+
+
+def main() -> int:
+    """Main entrypoint for running training jobs"""
+    parser = argparse.ArgumentParser(description=__doc__)
+
+    # Required Args
+    parser.add_argument(
+        "--training-kwargs",
+        "-k",
+        required=True,
+        help="Json string or json file pointer with keyword args for the training job",
+    )
+    parser.add_argument(
+        "--module",
+        "-m",
+        required=True,
+        help="Module name (package.Class) or UID to train",
+    )
+    parser.add_argument(
+        "--model-name",
+        "-n",
+        required=True,
+        help="Name to save the model under",
+    )
+
+    # Optional args
+    parser.add_argument(
+        "--save-path",
+        "-s",
+        default=".",
+        help="Path to save the output model to",
+    )
+    parser.add_argument(
+        "--library",
+        "-l",
+        nargs="*",
+        help="Libraries that need to be imported to register the module to train",
```

Comment:
nit: `Caikit libraries` ?

Reply:
Not sure that needs to be specified? I think the most correct term would be `Python libraries` since it is importing arbitrary python libs.